src/Illuminate/Database/UrlParser.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Illuminate\Database;
+
+use function array_map;
+use function parse_str;
+use function parse_url;
+use function array_merge;
+use function preg_replace;
+use Illuminate\Support\Arr;
+
+class UrlParser
+{
+    private static $driverAliases = [
+        'mssql' => 'sqlsrv',
+        'mysql2' => 'mysql', // Amazon RDS, for some weird reason
+        'postgres' => 'pgsql',
+        'postgresql' => 'pgsql',
+        'sqlite3' => 'sqlite',
+    ];
+
+    /**
+     * @var array
+     */
+    private $parsedUrl;
+
+    public static function getDriverAliases(): array
+    {
+        return self::$driverAliases;
+    }
+
+    public static function addDriverAlias(string $alias, string $driver)
+    {
+        self::$driverAliases[$alias] = $driver;
+    }
+
+    /**
+     * @param array|string $config
+     *
```

Comment:
Remove this whiteline and use proper spacing for @param annotations (two spaces before and after the types).

Reply:
StyleCI can't check this atm.